src/Microsoft.ML.Data/DataView/TypedCursor.cs

```diff
@@ -217,6 +217,18 @@ public RowCursor<TRow>[] GetCursorSet(Func<int, bool> additionalColumnsPredicate
                  .ToArray();
         }
 
+        public static void CheckTypeWithOnnx(Type userType, IDataView data)
+        {
```

Comment:
This has nothing to do with Onnx. How about calling this something like ValidateUserType?